main.js

```diff
@@ -45,12 +45,18 @@ async function run() {
   const octokit = new github.GitHub(token);
   const context = github.context;
 
+  let url;
   if (! context.payload.release) {
-    core.warning("Not running action as a release, skipping.");
-    return;
+      url = core.getInput('release-url', {required: false});
+      if(!url){
+        core.error("U need to provide an url if the action does not run on a release!");
```

Comment:
Good error checking here. I would like to stick with a warning instead of stopping for an error, mostly to handle the case where this was triggered by a draft.
```suggestion
        core.warning("No release URL, skipping. This action requires either a release URL passed in or run as part of a release event");
```